backend/user-service/src/app.service.ts

```diff
@@ -0,0 +1,40 @@
+import * as bcrypt from 'bcryptjs';
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { RpcException } from '@nestjs/microservices';
+import { User } from './schema/user.schema';
+import { CreateUserDto } from './dto';
+
+const SALT_ROUNDS = 10;
+
+@Injectable()
+export class AppService {
```

Comment:
We will need to add capabilities to add user profiling soon (e.g preferred languages, etc) as part of the onboarding process. 

Also, we will need to handle role-based access soon